EasyReflectometry/calculators/refl1d/wrapper.py

```diff
@@ -157,20 +156,21 @@ def calculate(self, x_array: np.ndarray, model_name: str) -> np.ndarray:
                     stack |= i.stack[j]
                 structure |= model.Repeat(stack, repeat=i.repeat.value)
 
-        argmin = np.argmin(x_array)
-        argmax = np.argmax(x_array)
-        dq_vector = x_array * self.storage['model'][model_name]['dq'] / 100 / (2 * np.sqrt(2 * np.log(2)))
+        argmin = np.argmin(q_array)
+        argmax = np.argmax(q_array)
+        dq_vector = self._resolution_function(q_array)
+        dq_vector_normalized_to_refnx = dq_vector * q_array / 100 / (2 * np.sqrt(2 * np.log(2)))
 
         q = names.QProbe(
-            x_array,
-            dq_vector,
+            q_array,
+            dq_vector_normalized_to_refnx,
             intensity=self.storage['model'][model_name]['scale'],
             background=self.storage['model'][model_name]['bkg'],
         )
         q.calc_Qo = np.linspace(
-            x_array[argmin] - 3.5 * dq_vector[argmin],
-            x_array[argmax] + 3.5 * dq_vector[argmax],
-            21 * len(x_array),
+            q_array[argmin] - 3.5 * dq_vector_normalized_to_refnx[argmin],
+            q_array[argmax] + 3.5 * dq_vector_normalized_to_refnx[argmax],
+            21 * len(q_array),
```

Comment:
Those hard-coded values `3.5` and `21` are very mysterious. Can we get some explanation or maybe make them parameters, defined somewhere at the top of the file?

Reply:
done